src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -258,32 +259,64 @@ public SeqColComparisonResultEntity compareSeqCols(
+        for (String element: commonSeqColAttributesValues) {
             Integer commonElementsCount = getCommonElementsCount(seqColAEntityMap.get(element), seqColBEntityMap.get(element));
             comparisonResult.putIntoElements("a-and-b", element, commonElementsCount);
         }
 
         // "elements" attribute | "a-and-b-same-order"
-        for (String attribute: commonSeqColAttributes) {
+        for (String attribute: commonSeqColAttributesValues) {
             if (lessThanTwoOverlappingElements(seqColAEntityMap.get(attribute), seqColBEntityMap.get(attribute))
                     || unbalancedDuplicatesPresent(seqColAEntityMap.get(attribute), seqColBEntityMap.get(attribute))){
                 comparisonResult.putIntoElements("a-and-b-same-order", attribute, null);
             } else {
-                boolean attributeSameOrder = seqColAEntityMap.get(attribute).equals(seqColBEntityMap.get(attribute));
+                boolean attributeSameOrder = check_A_And_B_Same_Order(seqColAEntityMap.get(attribute), seqColBEntityMap.get(attribute));
                 comparisonResult.putIntoElements("a-and-b-same-order", attribute, attributeSameOrder);
             }
         }
 
         return comparisonResult;
     }
 
+    /**
+     * Check whether the array of elements of elementsA are in the same order as the ones in elementsB.
+     * Example 1:
+     *      A = ["ch1", "B", "ch2", "ch3"],
+     *      B = ["ch1", "A", "ch2", "ch3"],
+     *      Common = ["ch1", "ch2", "ch3"] # Common elements between A and B
+     *      ==> A: indexOf("ch1") < indexOf("ch2") < indexOf("ch3") and B: indexOf("ch1") < indexOf("ch2") < indexOf("ch3")
+     *      ==> Same order elements
+     * Example 1:
+     *      A = ["ch1", "B", "ch2", "ch3"]
+     *      B = ["A", "ch1", "ch2", "ch3"]
+     *      Common = ["ch1", "ch2", "ch3"]
+     *      ==> A: indexOf("ch1") < indexOf("ch2") < indexOf("ch3") and B: indexOf("ch1") < indexOf("ch2") < indexOf("ch3")
+     *      ==> Same order elements
+     * NOTE: Assuming that the method List.retainAll() preserves the order in the original list (no counterexample at the moment)
+     * @see "https://github.com/ga4gh/seqcol-spec/blob/master/docs/decision_record.md#same-order-specification" */
+    public boolean check_A_And_B_Same_Order(List<String> elementsA, List<String> elementsB) {
+        LinkedList<String> elementsALocal = new LinkedList<>(elementsA);
+        LinkedList<String> elementsBLocal = new LinkedList<>(elementsB);
+        List<String> commonElements = getCommonElementsDistinct(elementsALocal, elementsBLocal);
+        elementsALocal.retainAll(commonElements); // Leaving only the common elements (keeping the original order to check)
+        elementsBLocal.retainAll(commonElements); // Leaving only the common elements (keeping the original order to check)
+        for (String element: commonElements) {
```

Comment:
I think once you've done `retainAll` (assuming the order is preserved), you should just be able to rely on list equality for the rest - i.e. just `return elementsALocal.equals(elementsBLocal);`

Reply:
True, I'll do that.